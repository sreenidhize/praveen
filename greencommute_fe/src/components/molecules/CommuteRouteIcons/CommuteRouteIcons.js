import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";
import CommuteIcon from "../../atom/CommuteIcons/CommuteIcons";

const useStyles = makeStyles((theme) => ({
  root: {
    fontSize: "13px",
    color: theme.typography.h2.color,
    marginTop: "20px",
    marginBottom: "8px",
  },
  display: {
    display: "flex",
  },
}));

const CommuteRoutes = (props) => {
  const classes = useStyles();
  const { icons, onClick } = props;

  return (
    <div data-testid="commuteRoutes">
      <div className={classes.display}>
        {icons.map((item) => {
          return (
            <CommuteIcon
              type={item.type}
              key={item.type}
              selected={item.selected}
              onClick={onClick}
            />
          );
        })}
      </div>
    </div>
  );
};

CommuteRoutes.propTypes = {
  icons: PropTypes.array,
  onClick: PropTypes.func,
};

export default CommuteRoutes;
