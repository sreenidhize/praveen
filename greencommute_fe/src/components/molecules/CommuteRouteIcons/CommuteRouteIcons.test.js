import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CommuteRouteIcons from './CommuteRouteIcons'

const icons = [
    {type : 'metro'},
    {type : 'bus'}
];

describe("CommuteRouteIcons component test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <CommuteRouteIcons icons={icons} />
    );
    expect(container).toMatchSnapshot();
  });

  test("test CommuteRouteIcons render", () => {
    const { getByTestId } = render(<CommuteRouteIcons icons={icons}/>);
    expect(getByTestId("commuteRoutes")).toBeInTheDocument();
  });
});