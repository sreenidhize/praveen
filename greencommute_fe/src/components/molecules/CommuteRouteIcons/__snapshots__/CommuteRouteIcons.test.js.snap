// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommuteRouteIcons component test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="commuteRoutes"
  >
    <div
      class="makeStyles-display-2"
    >
      <div>
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-7 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-root-3 makeStyles-iconUnselected-4"
            data-testid="metro"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="8.5"
              cy="14.5"
              r="1.5"
            />
            <circle
              cx="15.5"
              cy="14.5"
              r="1.5"
            />
            <path
              d="M12 2c-4 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h2l2-2h4l2 2h2v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-4-4-8-4zm0 2c3.51 0 4.96.48 5.57 1H6.43c.61-.52 2.06-1 5.57-1zM6 7h5v3H6V7zm12 8.5c0 .83-.67 1.5-1.5 1.5h-9c-.83 0-1.5-.67-1.5-1.5V12h12v3.5zm0-5.5h-5V7h5v3z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-7 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-root-3 makeStyles-iconUnselected-4"
            data-testid="bus"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 16c0 .88.39 1.67 1 2.22V20c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h8v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.78c.61-.55 1-1.34 1-2.22V6c0-3.5-3.58-4-8-4s-8 .5-8 4v10zm3.5 1c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm9 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm1.5-6H6V6h12v5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
